lib/actions/auth.coffee

```diff
@@ -134,11 +134,12 @@ exports.signup =
 				visuals.patterns.register(callback)
 
 			(credentials, callback) ->
-				resin.auth.register credentials, (error, token) ->
-					return callback(error, credentials)
+				resin.auth.register(credentials).then (token) ->
```

Comment:
You can use Bluebird's `.returns()` here: https://github.com/petkaantonov/bluebird/blob/master/API.md#returndynamic-value---promise.